crates/core/src/db/datastore/traits.rs

```diff
@@ -292,6 +293,73 @@ pub struct TableDef {
     pub(crate) table_access: StAccess,
 }
 
+impl TableDef {
```

Comment:
I don't love this code being here. But I don't think there's a great place to put it when we are in this state of defining a pile of concrete types, and traits (that aren't really used, relationaldb directly makes a Locking or whatever). Curious how you feel about it.

Reply:
Yeah, I think there's better ways to go about this, but ideally that would come with a general reorganizing of all these types.

Reply:
I also don't love this code being here FWIW